modules/examples/eb-sf-batch/deployspec.yaml

```diff
@@ -0,0 +1,35 @@
+deploy:
+  phases:
+    install:
+      commands:
+      - npm install -g aws-cdk@2.44.0
+      - pip install -r requirements.txt
+    build:
+      commands:
+      - aws iam create-service-linked-role --aws-service-name ecs-tasks.amazonaws.com || true
+      - aws iam create-service-linked-role --aws-service-name batch.amazonaws.com || true
+      - aws iam create-service-linked-role --aws-service-name spot.amazonaws.com || true
+      - aws iam create-service-linked-role --aws-service-name spotfleet.amazonaws.com || true
+      # - echo Building the Docker image...
+      - export ECR_REPO_NAME="addf-${ADDF_DEPLOYMENT_NAME}-${ADDF_MODULE_NAME}-${ADDF_PARAMETER_ECR_REPO_NAME}"
+      - export COMMIT_HASH=$(echo $CODEBUILD_RESOLVED_SOURCE_VERSION | cut -c 1-7)
+      - export IMAGE_TAG=${COMMIT_HASH:=latest}
+      - export REPOSITORY_URI=$AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$ECR_REPO_NAME
+      - aws ecr describe-repositories --repository-names ${ECR_REPO_NAME} || aws ecr create-repository --repository-name ${ECR_REPO_NAME}
+      - aws ecr get-login-password --region $AWS_DEFAULT_REGION | docker login --username AWS --password-stdin $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com
+      - cd images/simulation-mock && docker build -t $REPOSITORY_URI:latest .
+      - docker tag $REPOSITORY_URI:latest $REPOSITORY_URI:$IMAGE_TAG
+      - docker push $REPOSITORY_URI:latest && docker push $REPOSITORY_URI:$IMAGE_TAG
+      - cd ../../
+      - cdk deploy --all --require-approval never --progress events --app "python app.py" --outputs-file ./cdk-exports.json
+      # # Here we export some env vars and the write values into the batch_dag_config.py file for use by the DAGs
+      - export ADDF_MODULE_METADATA=$(python -c "import json; file=open('cdk-exports.json'); print(json.load(file)['addf-${ADDF_DEPLOYMENT_NAME}-${ADDF_MODULE_NAME}']['metadata'])")
+destroy:
+  phases:
+    install:
+      commands:
+      - npm install -g aws-cdk@2.44.0
+      - pip install -r requirements.txt
+    build:
+      commands:
+      - cdk destroy --force --app "python app.py"
```

Comment:
Do we need to clean up ECR on destroy? Any reason we don't create ECR via CDK?

Reply:
Good catch. I forgot to clean up ECR on destroy. 

Also, we can create ECR repos from CDK itself, but i'd be inclined to create a repo via deployspec, because i can run the below steps in a clean order (if need be):

`Create a repo -> build an image -> scan the image (using twistlock or aws inspector or ecr scanning) -> implement custom automation to decide if i want to push to ECR or not based on vulnerability findings` and then move on to the CDK stack.